### PR TITLE
Feature/workouts views

### PR DESCRIPTION
**Pull request by yudjinn**
changes some of the db models, added workout Create, Update, Delete, List, and Detail views. Made Workouts only show those of the logged in user.

### CODE REVIEW

File: onefit/dynamic_tz/apps.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from django.apps import AppConfig
+
+
+class DynamicTzConfig(AppConfig):
```
Comment (ambid17) [inline onefit/dynamic_tz/apps.py:4]: what is tz?

Comment (ambid17) [reply onefit/dynamic_tz/apps.py:4]: i see that its timezone, but it doesn't seem verbose enough

File: onefit/dynamic_tz/middleware.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import pytz
+
+from django.utils import timezone
+
+class TimezoneMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        tzname = request.session.get('django_timezone')
```
Comment (ambid17) [inline onefit/dynamic_tz/middleware.py:10]: needs to be a constant somewhere

**Review by ambid17 — APPROVED** [review]

File: onefit/dynamic_tz/apps.py (hunk 1 above)
Comment (yudjinn) [reply onefit/dynamic_tz/apps.py:4]: oh, would you prefer dynamic_timezone as the middleware name?

File: onefit/dynamic_tz/middleware.py (hunk 2 above)
Comment (yudjinn) [reply onefit/dynamic_tz/middleware.py:10]: the backend all runs in utc, this middleware is just to get the client's timezone when they access the site.

**Review by 0xLeif — APPROVED** [review]
💯

**Pull request merged**